gsdk/src/storage.rs

```diff
@@ -64,7 +64,7 @@ impl Api {
     }
 
     /// Get program pages from program id.
-    pub async fn program_pages(&self, pid: H256) -> Result<types::GearPages> {
+    pub async fn program_pages(&self, pid: ProgramId) -> Result<types::GearPages> {
```

Comment:
I think it is better to have more specific types whenever it is possible rather than abstract H256 or arrays

Reply:
What kind of efficiency do we get here when we use `H256` in lieu of `ProgramId`. I do not see any except for confusion.

Reply:
I see no efficiency improvements to use H256 as well
The only thing that for user lib we need use gear_core::ProgramId, while for gsdk we can go with metadata_generated::ProgramId

Reply:
I would go with `gear_core::*` in gsdk as well as this dependency is already there and as @clearloop metioned it will allow us to get rid of at least some types duplicated in metadata which in turn will minimize maintenance burden